src/composition/runner.rs

```diff
@@ -194,6 +194,15 @@ impl SubgraphWatchers {
     }
 }
 
+/// Events about watched subgraphs. If they're changed, the subgraph's name and changed SDL are
+/// emitted via SubgraphChanged. If they're removed, a SubgraphRemoved event is emitted with the
+/// name of the subgraph
+pub enum SubgraphEvent {
+    /// A change to the watched subgraph
+    SubgraphChanged(SubgraphChanged),
```

Comment:
I think this should probably be `SubgraphSchemaChanged`. `SubgraphWatchers` should be watching the `SupergraphConfig` file for changes and informing composition that the schema was either changed or removed.

Subgraphs being added amounts to a SubgraphSchemaChanged event for the purpose of composition, as it is just upserting sdls into its internal BTreeMap.

Reply:
I think we're on the same page; the watchers emit `SubgraphSchemaChanged` events, and in the tokio tasks we spin up for those watches, we send a `SubgraphChanged` event to make sure we're capturing the different kinds of events that can trigger recomposition (including subgraph removals--there's no SDL to emit in that case, and instead of marking it optional we're emitting a different event, `SubgraphRemoved`)